diff --git a/src/game/SpellEffects.cpp b/src/game/SpellEffects.cpp
index 23020ca..11a4277 100644
--- a/src/game/SpellEffects.cpp
+++ b/src/game/SpellEffects.cpp
@@ -336,6 +336,11 @@ void Spell::EffectSchoolDMG(uint32 effect_idx)
                         damage /= count;                    // divide to all targets
                         break;
                     }
+					case 29879:
+					{
+						damage = (unitTarget->GetMaxHealth()*26)/100;
+						break;
+					}
                     // percent from health with min
                     case 25599:                             // Thundercrash
                     {
